main/src/main/java/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -883,7 +904,44 @@ public void updateGeoData(final GeoData geo) {
                 activity.cacheDistanceView.setText(Units.getDistanceFromKilometers(geo.getCoords().distanceTo(activity.cache.getCoords())));
                 activity.cacheDistanceView.bringToFront();
             }
+            if (activity.isLiveDetail) {
+                if (activity.compassMiniView != null && activity.compassMiniView == activity.findViewById(R.id.compass_action)) {
+                    activity.setActualCoordinates(geo.getCoords());
+                } else if (activity.cache != null && activity.cache.getCoords() != null) {
+                    activity.initCompassMiniView();
+                }
+            }
+        }
+
+        @Override
+        public void updateDirection(final float direction) {
```

Comment:
Remove this method. Instead above, replace method `updateGeoData `with method `public void updateGeoDir(@NonNull final GeoData newGeo, final float newDirection) ` The new one can handle direction and geoposition changes in one place instead of two.